pkg/processor/metricfrequencyprocessor/processor_test.go

```diff
@@ -0,0 +1,211 @@
+package metricfrequencyprocessor
+
+import (
+	"context"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+func TestSieveAllFromEmpty(t *testing.T) {
+	ctx := context.Background()
+	sieve := &siftAllSieve{}
+	processor := &metricsfrequencyprocessor{sieve: sieve}
+
+	input := createMetrics()
+	result, err := processor.ProcessMetrics(ctx, input)
+
+	assert.NoError(t, err)
+	assert.Equal(t, result.ResourceMetrics().Len(), 0)
+}
+
+func TestSieveAllFromNonempty(t *testing.T) {
+	ctx := context.Background()
+	sieve := &siftAllSieve{}
+	processor := &metricsfrequencyprocessor{sieve: sieve}
+
+	resource1metrics := map[string][]string{
+		"lib-1": {"m1", "m2"},
+		"lib-2": {"m1", "m3"},
+	}
+	resource2metrics := map[string][]string{
+		"lib-1": {"m1", "m2"},
+		"lib-3": {"m1", "m3"},
+		"lib-4": {"m2", "m3"},
+	}
+	input := createMetrics(resource1metrics, resource2metrics)
+	result, err := processor.ProcessMetrics(ctx, input)
+
+	assert.NoError(t, err)
+	assert.Equal(t, result.ResourceMetrics().Len(), 0)
+}
+
+func TestKeepAllFromEmpty(t *testing.T) {
+	ctx := context.Background()
+	sieve := &keepAllSieve{}
+	processor := &metricsfrequencyprocessor{sieve: sieve}
+
+	input := createMetrics()
+	result, err := processor.ProcessMetrics(ctx, input)
+
+	assert.NoError(t, err)
+	assert.Equal(t, result.ResourceMetrics().Len(), 0)
+}
+
+func TestKeepAllFromNonEmpty(t *testing.T) {
+	ctx := context.Background()
```

Comment:
nit: it's just used once, I guess no need to assign it to a variable, just use it in place 🤷

Reply:
good idea, changed